extensions/microsoft-authentication/src/AADHelper.ts

```diff
@@ -833,6 +840,12 @@ export class AzureActiveDirectoryService {
 				Logger.error('session not found that was apparently just added');
 				return;
 			}
+
+			if (!this.sessionMatchesEndpoint(session)) {
```

Comment:
This change makes sense, I'm assuming there's not a similar check on the "remove" side because it will just not be found? It might be better to be explicit.

Reply:
By "remove side" do you mean these lines in `checkForUpdates()`? https://github.com/microsoft/vscode/blob/0dad6aec6d1d274d57d9d0a7eab1a4c75ca35ae8/extensions/microsoft-authentication/src/AADHelper.ts#L877-L880

Or these?
https://github.com/microsoft/vscode/blob/0dad6aec6d1d274d57d9d0a7eab1a4c75ca35ae8/extensions/microsoft-authentication/src/AADHelper.ts#L389-L405

Reply:
The first code block you linked.

Reply:
Ok, added a check inside the loop:

```typescript
		for (const { value } of e.removed) {
			if (!this.sessionMatchesEndpoint(value)) {
				// If the session wasn't made for this login endpoint, ignore this update
				continue;
			}

			Logger.info(`Session removed in another window with scopes: ${value.scope}`);
			await this.removeSessionById(value.id, false);
		}
```